### DISCUSSION TITLE
RuntimeError: "slow_conv2d_cpu" not implemented for 'Half'

### DISCUSSION THREAD

**Discussion by v-puppala** [openai/whisper#92] [General]
I ran the short code snippet in the readme and ran it on my own colab notebook. I received an error on the final line (result=whisper.decode(model, mel, options)

# Here's error message and traceback:

RuntimeError                              Traceback (most recent call last)
[<ipython-input-14-f06dd8c2b7e0>](https://localhost:8080/#) in <module>
     17 # decode the audio
     18 options = whisper.DecodingOptions()
---> 19 result = whisper.decode(model, mel, options)
     20 
     21 # print the recognized text

10 frames
[/usr/local/lib/python3.7/dist-packages/torch/nn/modules/conv.py](https://localhost:8080/#) in _conv_forward(self, input, weight, bias)
    302                             _single(0), self.dilation, self.groups)
    303         return F.conv1d(input, weight, bias, self.stride,
--> 304                         self.padding, self.dilation, self.groups)
    305 
    306     def forward(self, input: Tensor) -> Tensor:

RuntimeError: "slow_conv2d_cpu" not implemented for 'Half

**Comment by Batuhanapa** [comment]
18 options = whisper.DecodingOptions(fp16  = False), it worked for me

**Reply by kumbarisagar** [reply]
i actually got the same error because when i turned float 32 operations into float16 operations

**Reply by kumbarisagar** [reply]
how to solve it??\

**Reply by marinone94** [reply]
Are you running on CPU?

**Reply by Batuhanapa** [reply]
> how to solve it??\

İf u wanna use float16 type u should work on GPU

**Comment by kumbarisagar** [comment]
Yes , i am running on CPU on raspberry pi

**Reply by megatron100** [reply]
I am interested in your project. May I know why raspberry pi, are using it to record conversation?

**Comment by ooluwaseunabel** [comment]
I ran into the same error. I was able to resolve it after changing this line:
`options = whisper.DecodingOptions()`

to
`options = whisper.DecodingOptions(fp16 = False)`

**Reply by roydellclarke** [reply]
thanks dude, lifesaver

**Reply by roydellclarke** [reply]
worked! let's go put a dent in the universe.

**Reply by ooluwaseunabel** [reply]
> worked! let's go put a dent in the universe.

Glad to know it worked for you.

**Reply by sithart** [reply]
Thanks dude, It's working fine now.

**Reply by Randalf96** [reply]
thanks man it worked for me 2 !!!! life saver

**Reply by ideelahmed** [reply]
thank you for blessing us with your awesomeness. Sh*t works!

**Reply by StepanTrnkaCNC** [reply]
Where did you change those options, please?

**Reply by chazmofdoom** [reply]
Thank you! This is the solution!

**Reply by clevadani** [reply]
Worked like a charm.

**Comment by halqadasi** [comment]
I added this option (--fp32) and it works for me, for example:

`!python inference_realesrgan.py -n realesr-general-x4v3 -i upload -g 0 -dn 0.2 --fp32`

**Reply by andrez4vz** [reply]
Great! It's works. Thanks man!

**Comment by AbdulHaseeb007** [comment]
updating ultralytics to 8.0.109 also works, just in case u have changed your ultralytics version
